constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/ShiftBy.java

```diff
@@ -0,0 +1,60 @@
+package gov.nasa.jpl.aerie.constraints.tree;
+
+import gov.nasa.jpl.aerie.constraints.model.ActivityInstance;
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.time.Window;
+import gov.nasa.jpl.aerie.constraints.time.Windows;
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public final class ShiftBy implements Expression<Windows> {
+  public final Expression<Windows> windows;
+  public final Duration fromStart;
+  public final Duration fromEnd;
+
+  public ShiftBy(final Expression<Windows> left, final Duration clampFromStart, final Duration clampFromEnd) {
+    this.windows = left;
+    this.fromStart = clampFromStart;
+    this.fromEnd = clampFromEnd;
+  }
+
+  @Override
+  public Windows evaluate(final SimulationResults results, final Window bounds, final Map<String, ActivityInstance> environment) {
+    final var windows = this.windows.evaluate(results, bounds, environment);
+    return windows.shiftBy(this.fromStart, this.fromEnd);
+  }
+
+  @Override
+  public void extractResources(final Set<String> names) {
+    this.windows.extractResources(names);
+  }
+
+  @Override
+  public String prettyPrint(final String prefix) {
+    return String.format(
+        "\n%s(clamp %s by %s %s)",
```

Comment:
:small_red_triangle_down: Should this be "shiftBy" now that we've changed the method name?